src/sentry/static/sentry/app/stores/discoverSavedQueriesStore.tsx

```diff
@@ -88,7 +109,8 @@ const DiscoverSavedQueriesStore = Reflux.createStore({
     let savedQueries;
     const index = this.state.savedQueries.findIndex(item => item.id === query.id);
     if (index > -1) {
-      savedQueries = [...this.state.savedQueries].splice(index, 1, query);
+      savedQueries = [...this.state.savedQueries];
+      savedQueries.splice(index, 1, query);
```

Comment:
Do we have a test for this?

Reply:
We do! I added a test to the store spec that fails with the old code.